src/main/java/de/tum/in/www1/artemis/web/rest/AccountResource.java

```diff
@@ -205,11 +205,11 @@ public void changePassword(@RequestBody PasswordChangeDTO passwordChangeDto) {
     /**
      * {@code POST   /account/reset-password/init} : Send an email to reset the password of the user.
      *
-     * @param mail the mail of the user.
+     * @param mailUsername string containing either mail or username of the user.
      */
     @PostMapping(path = "/account/reset-password/init")
-    public void requestPasswordReset(@RequestBody String mail) {
-        List<User> user = userRepository.findAllByEmailIgnoreCase(mail);
+    public void requestPasswordReset(@RequestBody String mailUsername) {
+        List<User> user = userRepository.findAllByEmailOrUsernameIgnoreCase(mailUsername);
         if (!user.isEmpty()) {
             List<User> internalUsers = user.stream().filter(User::isInternal).toList();
```

Comment:
Maybe an edge case. It might be possible that one email is associated to multiple users ? (Login should be a Primary Key in the DB. Mail may be not)

Maybe you should reset the password for all matched users or warn that there are multiple?

Reply:
It's not a unique key, correct. But I just checked and duplication is not possible. Email addresses get checked for duplication on registration, creation, update and LTI init.